Source/CarthageKit/Project.swift

```diff
@@ -579,49 +575,42 @@ public final class Project { // swiftlint:disable:this type_body_length
-		return SignalProducer<Bool, CarthageError>(value: self.useBinaries)
-			.flatMap(.merge) { useBinaries -> SignalProducer<Bool, CarthageError> in
-				if !useBinaries {
-					return SignalProducer(value: false)
-				}
-
-				let checkoutDirectoryURL = self.directoryURL.appendingPathComponent(dependency.relativePath, isDirectory: true)
-
-				switch dependency {
-				case let .gitHub(server, repository):
-					let client = Client(server: server)
-					return self.downloadMatchingBinaries(for: dependency, pinnedVersion: pinnedVersion, fromRepository: repository, client: client)
-						.flatMapError { error -> SignalProducer<URL, CarthageError> in
-							if !client.isAuthenticated {
-								return SignalProducer(error: error)
-							}
-							return self.downloadMatchingBinaries(
-								for: dependency,
-								pinnedVersion: pinnedVersion,
-								fromRepository: repository,
-								client: Client(server: server, isAuthenticated: false)
-							)
-						}
-						.flatMap(.concat) {
-							return self.unarchiveAndCopyBinaryFrameworks(zipFile: $0, projectName: dependency.name, pinnedVersion: pinnedVersion, toolchain: toolchain)
-								.on(value: { _ in
-									// Trash the checkouts folder when we've successfully copied binaries, to avoid building unnecessarily
-									_ = try? FileManager.default.trashItem(at: checkoutDirectoryURL, resultingItemURL: nil)
-								})
-						}
-						.flatMap(.concat) { self.removeItem(at: $0) }
-						.map { true }
-						.flatMapError { error in
-							self._projectEventsObserver.send(value: .skippedInstallingBinaries(dependency: dependency, error: error))
-							return SignalProducer(value: false)
-						}
-						.concat(value: false)
-						.take(first: 1)
+		let checkoutDirectoryURL = self.directoryURL.appendingPathComponent(dependency.relativePath, isDirectory: true)
 
-				case .git, .binary:
+		switch dependency {
+		case let .gitHub(server, repository):
+			let client = Client(server: server)
+			return self.downloadMatchingBinaries(for: dependency, pinnedVersion: pinnedVersion, fromRepository: repository, client: client)
+				.flatMapError { error -> SignalProducer<URL, CarthageError> in
+					if !client.isAuthenticated {
+						return SignalProducer(error: error)
+					}
+					return self.downloadMatchingBinaries(
+						for: dependency,
+						pinnedVersion: pinnedVersion,
+						fromRepository: repository,
+						client: Client(server: server, isAuthenticated: false)
+					)
+				}
+				.flatMap(.concat) {
+					return self.unarchiveAndCopyBinaryFrameworks(zipFile: $0, projectName: dependency.name, pinnedVersion: pinnedVersion, toolchain: toolchain)
+						.on(value: { _ in
+							// Trash the checkouts folder when we've successfully copied binaries, to avoid building unnecessarily
+							_ = try? FileManager.default.trashItem(at: checkoutDirectoryURL, resultingItemURL: nil)
```

Comment:
We should get rid of this.

The current behavior is to delete the checkout for a dependency after successfully using a downloaded binary. This exists because previously the existence of the checkout directory determined whether or not that dependency was built.

Now that downloading binaries is part of build, we should get right of that line to delete the checkout. It looks like `buildCheckedOutDependenciesWithOptions` can already handle this.